### ISSUE TITLE
Result iteration on postgres requires a transaction

### ISSUE THREAD

**Issue by sm-Fifteen** [encode/databases#102]
I'm not sure if this is meant to be the intended behavior or not, since iterate() isn't very documented as of right now, but attempting to use it without creating a transaction first causes asyncpg to throw the following: `asyncpg.exceptions.NoActiveSQLTransactionError: cursor cannot be created outside of a transaction`.

The issue can easily be reproduced with the following snippet. I haven't tested other backends to see if the same thing occurs with SQLite and MySQL.

```py
from databases import Database
import asyncio

mydb = Database('postgresql://localhost:5433/testdb11')

async def test():
    await mydb.connect()

    async for row in mydb.iterate(query='SELECT * FROM (VALUES (1), (2), (3), (4), (5)) as t(num)'):
        print(row)

    await mydb.disconnect()

loop = asyncio.get_event_loop()  
loop.run_until_complete(test())  
loop.close()
```

SQLAlchemy doesn't require the use of transactions to iteratively load the results of a query, and the documentation doesn't specify this constraint, so transactions probably shouldn't be enforced here either as far as the public API for functions goes.

**Comment by tomchristie** [comment]
Okay. I guess we could implicitly create a transaction within the postgres driver code if there isn't already one exsiting.

**Comment by tomchristie** [comment]
Best thing to help progress this issue would be a pull request with a failing test case.
That'd also help us identify if we have the same issue against other drivers or not.

**Comment by tomchristie** [comment]
Closed via #111 - We now just always wrap iteations up inside a transaction, whatever. (Which may be a sub-transaction/savepoint) Seems pretty reasonable.
